ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -946,3 +891,416 @@ def test_ipa_hbac_rule_duplication(self):
+        cmd1 = self.replicas[0].run_command(
+            ["ipa", "sudorule-find", sudorule])
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "sudocmd-find", sudocmd])
+        assert 'Rule name: readfiles\n' in cmd1.stdout_text
+        assert 'Sudo Command: /usr/bin/less\n' in cmd2.stdout_text
+
+    def test_ipa_migrate_prod_mode_new_user_sid(self):
+        """
+        This testcase checks that in prod-mode uid/gid of the
+        migrated user is preserved i.e we have same
+        uid/gid for user on remote and local IPA server.
+        """
+        username = 'produser1'
+        original_passwd = 'Secret123'
+        tasks.user_add(self.master, username, password=original_passwd)
+        run_migrate(
+            self.replicas[0],
+            "stage-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n'],
+        )
+        tasks.clear_sssd_cache(self.master)
+        result1 = self.master.run_command(['id', username])
+        tasks.clear_sssd_cache(self.replicas[0])
+        result2 = self.replicas[0].run_command(['id', username])
+        assert result1.stdout_text == result2.stdout_text
+
+
+class TestIPAMigrationWithADtrust(IntegrationTest):
+    """
+    Test for ipa-migrate tool with IPA Master having trust setup
+    with Windows AD.
+    """
+    topology = "line"
+    num_ad_domains = 1
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(
+            cls.master, setup_dns=True, extra_args=['--no-dnssec-validation']
+        )
+        cls.ad = cls.ads[0]
+        cls.ad_domain = cls.ad.domain.name
+        tasks.install_adtrust(cls.master)
+        tasks.configure_dns_for_trust(cls.master, cls.ad)
+        tasks.establish_trust_with_ad(cls.master, cls.ad.domain.name)
+
+    def test_install_local_server(self):
+        """
+        This test installs local IPA Server() i.e new IPA server with
+        the same realm and domain name that will receive the migration data.
+        """
+        tasks.install_master(self.replicas[0], setup_dns=True, setup_kra=True)
+        tasks.install_adtrust(self.replicas[0])
+
+    def test_check_ad_attributes_migrate_prod_mode(self):
+        """
+        This test checks that IPA-AD trust related attributes
+        are migrated to local server.
+        """
+        base_dn = str(self.master.domain.basedn)
+        TRUST_LOG = (
+            "Added entry: cn={},cn=ad,cn=trusts,{}"
+        ).format(self.ad_domain, base_dn)
+        CMD_LOG = "Trust direction: Trusting forest"
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd = self.replicas[0].run_command(
+            ['ipa', 'trust-show', self.ad_domain]
+        )
```

Comment:
You can do instead:
```
trust1 = self.master.run_command(['ipa', 'trust-show', self.ad_domain]).stdout_text
trust2 = self.replicas[0].run_command(['ipa', 'trust-show', self.ad_domain]).stdout_text
assert trust1 == trust2
```
This would ensure that all the attributes have been migrated with an identical value.